lib/Dialect/IR/TcpOps.cpp

```diff
@@ -170,6 +176,31 @@ LogicalResult CastOp::verify() {
   return success();
 }
 
+LogicalResult GatherOp::verify() {
+  auto inputTensor = cast<RankedTensorType>(getInput().getType());
+  auto indicesTensor = cast<RankedTensorType>(getIndices().getType());
+  int64_t gatherDim = getDimAttr().getValue().getSExtValue();
+
+  if (inputTensor.getRank() != indicesTensor.getRank())
+    return emitOpError("tcp.gather requires that the input tensor and indices "
+                       "are the same rank");
+
+  for (int i = 0; i < inputTensor.getRank(); i++) {
+    if (inputTensor.getShape()[i] != indicesTensor.getShape()[i] &&
+        !(inputTensor.getShape()[i] == ShapedType::kDynamic ||
+          i == gatherDim)) {
+      return emitOpError("indices tensor does not match expected shape");
+    }
+  }
```

Comment:
This doesn't seem to match the semantics of `tcp.gather`, which should have the same constraints as `torch.gather`.

Specifically the following from [here](https://pytorch.org/docs/stable/generated/torch.gather.html#torch.gather)
> It is also required that index.size(d) <= input.size(d) for all dimensions d != dim

Reply:
I updated it